packages/flutter/test/material/navigation_rail_test.dart

```diff
@@ -2022,6 +2022,80 @@ void main() {
 
     semantics.dispose();
   });
+
+  testWidgets('NavigationRailDestination padding properly applied', (WidgetTester tester) async {
+    const EdgeInsets firstItemPadding = EdgeInsets.all(20);
+    const EdgeInsets secondItemPadding = EdgeInsets.symmetric(vertical: 30);
+    const EdgeInsets thirdItemPadding = EdgeInsets.symmetric(horizontal: 10);
+
+    await _pumpNavigationRail(
+      tester,
+      navigationRail: NavigationRail(
+        selectedIndex: 0,
+        destinations: const <NavigationRailDestination>[
+          NavigationRailDestination(
+            icon: Icon(Icons.favorite_border),
+            selectedIcon: Icon(Icons.favorite),
+            label: Text('Abc'),
+            padding: firstItemPadding,
+          ),
+          NavigationRailDestination(
+            icon: Icon(Icons.bookmark_border),
+            selectedIcon: Icon(Icons.bookmark),
+            label: Text('Def'),
+            padding: secondItemPadding,
+          ),
+          NavigationRailDestination(
+            icon: Icon(Icons.star_border),
+            selectedIcon: Icon(Icons.star),
+            label: Text('Ghi'),
+            padding: thirdItemPadding
+          ),
+        ],
+        labelType: NavigationRailLabelType.all,
```

Comment:
place this parameter above `navigationRail`, since it's easy to miss and we prefer any parameter that is the `child` Widget (or similar) to be the last param